common/src/main/java/io/homeassistant/companion/android/common/sensors/NetworkSensorManager.kt

```diff
@@ -168,6 +184,7 @@ class NetworkSensorManager : SensorManager {
     override fun requestSensorUpdate(
         context: Context
     ) {
+        updateHotspotEnabledSensor(context)
```

Comment:
None of the checks done for this sensor is required for the location permission, you can skip that for this sensor in the `requiredPermissions()` method